lib/blueprinter/base.rb

```diff
@@ -225,22 +240,25 @@ def self.render_as_json(object, options= {})
     # so we rename it for clarity
     #
     # @api private
-    def self.prepare(object, view_name:, local_options:)
+    def self.prepare(object, view_name:, local_options:, root: nil, meta: nil)
       unless view_collection.has_view? view_name
         raise BlueprinterError, "View '#{view_name}' is not defined"
       end
       prepared_object = include_associations(object, view_name: view_name)
       if array_like?(object)
-        prepared_object.map do |obj|
+        data = prepared_object.map do |obj|
           object_to_hash(obj,
                          view_name: view_name,
                          local_options: local_options)
         end
       else
-        object_to_hash(prepared_object,
+        data = object_to_hash(prepared_object,
                        view_name: view_name,
                        local_options: local_options)
       end
+      return data unless root
```

Comment:
These three lines feel a bit sloppy and tacked on. `.prepare` is now doing potentially 7 separate things. I wouldn't feel comfortable merging something like this, but I would be more comfortable with making `.prepare` more functional and declarative... something like
```
self.prepare(args)
  raise view_not_defined unless view_defined?
  prepend_meta(prepend_root(shape_data(include_associations(args))))
end
```

I would much prefer several no-ops to complicated control flow in a bloated method.